pkgs/build-support/setup-hooks/auto-patchelf.py

```diff
@@ -264,10 +262,16 @@ def auto_patchelf(
     missing = [dep for dep in dependencies if not dep.found]
 
     # Print a summary of the missing dependencies at the end
+    print(f"auto-patchelf: {len(missing)} dependencies not be satisfied")
+    failure = False
     for dep in missing:
-        print(f"auto-patchelf could not satisfy dependency {dep.name} wanted by {dep.file}")
+        if dep.name.name in ignore_missing or "*" in ignore_missing:
```

Comment:
Since the "*" is not a regex anyway, we may as well use "true" as the catch-all, and keep the old semantics easily. Not very nice from a type viewpoint I will admit :-)

Reply:
I opted to handle the true -> [ "*" ] translation in the autoPatchelfHook layer (auto-patchelf.sh). I found that this provided the cleanest separation of concerns and API layers at each of the steps.